packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts

```diff
@@ -46,6 +52,74 @@ function validateKeyField(field: FieldDefinitionNode): void {
     throw new InvalidDirectiveError(`If you define a field and specify it as a 'keyField', it must be of type 'ID' or 'String'.`)
 }
 
+/**
+ * Ensure that the field passed in is compatible to be a key field
+ * (Not a list and of type ID or String)
+ * @param field: the field to be checked.
+ */
+function validateKeyFieldNewConnection(field: FieldDefinitionNode): void {
+    if (!field) {
+        return
+    }
+    const isNonNull = isNonNullType(field.type);
+    const isAList = isListType(field.type)
+
+    // The only valid key fields are single non-null fields.
+    if (!isAList && isNonNull) {
+        return;
+    }
+    throw new InvalidDirectiveError(`All fields provided to an @connection must be non-null scalar fields.`)
```

Comment:
Should we validate if the field type is scalar, since the error message suggest that we validate it.

Reply:
*scalar or enum.